src/pool/pool_jemalloc.c

```diff
@@ -37,6 +37,8 @@
 typedef struct jemalloc_memory_pool_t {
     umf_memory_provider_handle_t provider;
     unsigned int arena_index; // index of jemalloc arena
+    // set to true if umfMemoryProviderFree() should never be called
+    bool disable_provider_free;
```

Comment:
Can we replace this bool (similar to pool_jemalloc.h:23) with:
`umf_jemalloc_pool_params_t` params or sth?